Tests/ConfigCatTests/ConfigCatClientTests.swift

```diff
@@ -356,6 +369,166 @@ class ConfigCatClientTests: XCTestCase {
+
+        client.clearDefaultUser()
+
+        let expectation4 = self.expectation(description: "wait for response")
+        client.getValue(for: "key", defaultValue: "") { val in
+            XCTAssertEqual("def", val)
+            expectation4.fulfill()
+        }
+        wait(for: [expectation4], timeout: 2)
+    }
+
+    func testHooks() {
+        MockHTTP.enqueueResponse(response: Response(body: String(format: testJsonFormat, "\"fake\""), statusCode: 200))
+        MockHTTP.enqueueResponse(response: Response(body: "", statusCode: 500))
+        var error = ""
+        var changed = false
+        var ready = false
+        let hooks = Hooks()
+        hooks.addOnError { e in
+            error = e
+        }
+        hooks.addOnReady {
+            ready = true
+        }
+        hooks.addOnConfigChanged { _ in
+            changed = true
+        }
+        let client = ConfigCatClient(sdkKey: "test", refreshMode: PollingModes.manualPoll(), session: MockHTTP.session(), hooks: hooks)
+        let expectation = self.expectation(description: "wait for response")
+        client.forceRefresh { r in
+            XCTAssertTrue(r.success)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 2)
+        let expectation2 = self.expectation(description: "wait for response")
+        client.forceRefresh { r in
+            XCTAssertFalse(r.success)
+            expectation2.fulfill()
+        }
+        wait(for: [expectation2], timeout: 2)
+
+        waitFor {
+            changed && ready && error.starts(with: "Double-check your SDK Key at https://app.configcat.com/sdkkey.") && error.contains("500")
+        }
+    }
+
+    func testHooksSub() {
+        MockHTTP.enqueueResponse(response: Response(body: String(format: testJsonFormat, "\"fake\""), statusCode: 200))
+        MockHTTP.enqueueResponse(response: Response(body: "", statusCode: 500))
+        var error = ""
+        var changed = false
+        let hooks = Hooks()
+        let client = ConfigCatClient(sdkKey: "test", refreshMode: PollingModes.manualPoll(), session: MockHTTP.session(), hooks: hooks)
+        client.hooks.addOnError { e in
+            error = e
+        }
+        client.hooks.addOnConfigChanged { _ in
+            changed = true
+        }
+
+        let expectation = self.expectation(description: "wait for response")
+        client.forceRefresh { r in
+            XCTAssertTrue(r.success)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 2)
+        let expectation2 = self.expectation(description: "wait for response")
+        client.forceRefresh { r in
+            XCTAssertFalse(r.success)
+            expectation2.fulfill()
+        }
+        wait(for: [expectation2], timeout: 2)
+
+        waitFor {
+            changed && error.starts(with: "Double-check your SDK Key at https://app.configcat.com/sdkkey.") && error.contains("500")
+        }
+    }
+
+    func testOnFlagEvalError() {
```

Comment:
```suggestion
    func testOnFlagEvaluationError() {
```
I would use `Evaluation` instead of `Eval`